pkg/util/util.go

```diff
@@ -0,0 +1,114 @@
+func Getter(namespace string, clientset *kubernetes.Clientset, kinds string, c chan interface{}) {
+	defer close(c)
+	var err error
+	var list interface{}
+
+	if kinds != "" {
+		resources = strings.Split(kinds, ",")
+	}
+
+	for _, resource := range resources {
+		switch resource {
+		case "Pods":
+			list, err = clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "ComponentStatuses":
+			list, err = clientset.CoreV1().ComponentStatuses().List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "ConfigMaps":
+			list, err = clientset.CoreV1().ConfigMaps(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Endpoints":
+			list, err = clientset.CoreV1().Endpoints(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Events":
+			list, err = clientset.CoreV1().Events(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "LimitRanges":
+			list, err = clientset.CoreV1().LimitRanges(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Namespaces":
+			list, err = clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "PersistentVolumes":
+			list, err = clientset.CoreV1().PersistentVolumes().List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "PersistentVolumeClaims":
+			list, err = clientset.CoreV1().PersistentVolumeClaims(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "PodTemplates":
+			list, err = clientset.CoreV1().PodTemplates(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "ResourceQuotas":
+			list, err = clientset.CoreV1().ResourceQuotas(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Secrets":
+			list, err = clientset.CoreV1().Secrets(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Services":
+			list, err = clientset.CoreV1().Services(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "ServiceAccounts":
+			list, err = clientset.CoreV1().ServiceAccounts(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+
+		// these will be from the AppsV1
+		case "DaemonSets":
+			list, err = clientset.AppsV1().DaemonSets(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "Deployments":
+			list, err = clientset.AppsV1().Deployments(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "ReplicaSets":
+			list, err = clientset.AppsV1().ReplicaSets(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		case "StatefulSets":
+			list, err = clientset.AppsV1().StatefulSets(namespace).List(metav1.ListOptions{})
+			handleError(err, resource)
+		default:
+			log.Error("Given kind is not supported")
+			return
+		}
+
+		c <- list
```

Comment:
@arush-sal 
The only concern that I have is if we really need to send the list of channel, can it be done by just returning the list and not using routines, I would prefer that. 
I think even if don use routines you would be able to use this in another project.
But If you think using routines is better idea and have tested the things, please confirm, i will go ahead to merge this.

Reply:
@viveksinghggits I cannot think of a way through which we can return individual `ResourceList` type as we are iterate over them. The only other possible solution that I can think of is returning a slice with all the lists, then iterating over them. But that would make the overall execution blocking and would unnecessarily add a few more loops. As far as testing goes, yes, I do have tested the overall code both here(in execution) and in groot(in importing.). So I am pretty confident about this.